packages/dart_frog_cli/e2e/test/dev_test.dart

```diff
@@ -47,4 +47,69 @@ void main() {
       expect(response.body, 'Route not found');
     });
   });
+
+  group('dart_frog dev when ran multiple times', () {
+    const projectName1 = 'example1';
+    const projectName2 = 'example2';
+
+    final tempDirectory = Directory.systemTemp.createTempSync();
+
+    Process? process1;
+    Process? process2;
+
+    setUp(() async {
+      await dartFrogCreate(projectName: projectName1, directory: tempDirectory);
+      await dartFrogCreate(projectName: projectName2, directory: tempDirectory);
+    });
+
+    tearDown(() async {
+      if (process1 != null) {
+        killDartFrogServer(process1!.pid).ignore();
+      }
+      if (process2 != null) {
+        killDartFrogServer(process2!.pid).ignore();
+      }
+
+      // await tempDirectory1.delete(recursive: true);
+    });
+
+    test(
+      'Running 2 different dart_frog dev command will fail '
+      'when different dart vm port is not set',
+      () async {
+        process1 = await dartFrogDev(
+          directory: Directory(path.join(tempDirectory.path, projectName1)),
+        );
+
+        try {
+          await dartFrogDev(
+            directory: Directory(path.join(tempDirectory.path, projectName2)),
+            exitOnError: false,
+          );
+          fail('exception not thrown');
+        } catch (e) {
```

Comment:
Can we catch the specific exception type being thrown here? Thus, if the test fails with another exception the failure is not a "foo doesn't contain x" but instead the thrown exception.

Reply:
I agree. The thing is caught `e` value is not a type `Exception` just a `String`. What do you think is best we do here?

Reply:
Would something like the following work?

```dart
try {
// ...
} on String catch(e) {
// ...
} catch (e) {
// ..
}
```

or simply:

```dart
try {
// ...
} on String catch(e) {
// ...
}
```